k256/src/arithmetic/scalar.rs

```diff
@@ -65,6 +65,12 @@ impl Scalar {
         self.0.truncate_to_u32()
     }
 
+    /// Attempts to parse the given byte array as a scalar.
+    /// Does not check the result for being in the correct range.
+    pub const fn from_bytes_unchecked(bytes: &[u8; 32]) -> Self {
```

Comment:
Is there a specific reason why this needs to be `pub` as opposed to `pub(crate)`?

Reply:
Not really, I didn't think about what should be exposed to the user. This method is useful for defining `Scalar` constants, which I suppose may be needed sometimes? But if you want to preserve the current public API, this can be set to `pub(crate)` for now.

Reply:
Given that this method can violate the invariant of what a `Scalar` is supposed to contain, I'd suggest making it private for now.

I think ideally a `pub const fn from_bytes(...)` method would be possible.

Reply:
Yes, I think `from_bytes()` can be easily made into a `const fn` (by using the existing `unchecked` code). Probably can wait for another PR though.